src/snowflake/cli/_plugins/streamlit/streamlit_entity.py

```diff
@@ -1,12 +1,60 @@
+from pathlib import Path
+from typing import Optional
+
+from snowflake.cli._plugins.nativeapp.artifacts import build_bundle
+from snowflake.cli._plugins.nativeapp.feature_flags import FeatureFlag
 from snowflake.cli._plugins.streamlit.streamlit_entity_model import (
     StreamlitEntityModel,
 )
+from snowflake.cli._plugins.workspace.context import ActionContext
 from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli.api.project.schemas.v1.native_app.path_mapping import PathMapping
 
 
+# WARNING: This entity is not implemented yet. The logic below is only for demonstrating the
+# required interfaces for composability (used by ApplicationPackageEntity behind a feature flag).
 class StreamlitEntity(EntityBase[StreamlitEntityModel]):
     """
     A Streamlit app.
     """
 
-    pass
+    @property
+    def project_root(self) -> Path:
+        return self._workspace_ctx.project_root
+
+    @property
+    def deploy_root(self) -> Path:
+        return self.project_root / "output" / "deploy"
+
+    def _verify_feature_flag_enabled(self):
```

Comment:
I have it there as well. I added it here so that we won't use this fake entity via another path by mistake (reusing the same feature flag for simplicity).